### PR TITLE
cinder: Add Pure Storage FlashArray config to cinder barclamp

### PR DESCRIPTION
**Pull request by sdodsley**
Adds config to the UI, locale, cinder.conf, cinder schema
and add migration.

### CODE REVIEW

**Comment by jsuchome** [comment]
Please add "cinder: " prefix to your commit. For example "cinder: Add Pure Storage FlashArray config"

**Comment by sdodsley** [comment]
@jsuchome what's the next step in the process to get this reviewed?

File: chef/data_bags/crowbar/template-cinder.schema [hunk 1]
```diff
@@ -153,6 +153,12 @@
                   "hitachi_zoning_request": { "type" : "bool", "required" : true }
                  }
                 },
+                "pure": {
+                "type": "map", "mapping": {
+                    "san_ip": { "type": "str", "required": true },
+                    "pure_api_token": { "type": "str", "required": true },
+                    "use_chap_auth": { "type": "bool", "required": true }
+                }
```
Comment (jsuchome) [inline chef/data_bags/crowbar/template-cinder.schema]: `},` missing here

File: crowbar_framework/app/views/barclamp/cinder/_edit_attributes.html.haml [hunk 2]
```diff
@@ -192,6 +192,17 @@
...
+          {{/if_eq}}
+          {{#if_eq backend_driver 'pure'}}
+          %li.list-group-item
+            %fieldset
+              %legend
+                = t('.volumes.pure_parameters')
+
+              = string_field %w(volumes {{@index}} pure san_ip)
+              = string_field %w(volumes {{@index}} pure pure_api_token)
+              = boolean_field %w(volumes {{@index}} eqlx use_chap_auth)
```
Comment (jsuchome) [inline crowbar_framework/app/views/barclamp/cinder/_edit_attributes.html.haml]: `s/eqlx/pure`

Comment (sdodsley) [reply crowbar_framework/app/views/barclamp/cinder/_edit_attributes.html.haml]: Oops - missed one :)

**Comment by jsuchome** [comment]
Thanks @sdodsley ! 
This looks correct from crowbar POV (I was able to deploy it, even with SOC7). However I do not know about this backend, so I can't really judge if those openstack bits are correct... maybe @toabctl could take a look?

**Comment by sdodsley** [comment]
I work for Pure and know the backend and the Cinder driver. From an OpenStack perspective, these are the minimum correct parameters. More details are here: https://docs.openstack.org/cinder/queens/configuration/block-storage/drivers/pure-storage-driver.html

**Comment by vuntz** [comment]
> Anyway, now someone has to decide if (and/or when) do we want to include the support for Pure in the product.

My take is that it's fine to expose it.

File: crowbar_framework/config/locales/cinder/en.yml [hunk 3]
```diff
@@ -116,6 +118,10 @@ en:
               vserver: 'Name of the Virtual Storage Server (netapp_vserver)'
             nfs:
               nfs_shares: 'List of NFS Exports'
+            pure:
+              use_chap_auth: 'Use chap auth for targets'
+              san_ip: 'Pure Storage Flasharray Management VIP'
+              pure_api_token: 'Pure Storage flashArray API toekn'
```
Comment (vuntz) [inline crowbar_framework/config/locales/cinder/en.yml]: Ah, just one thing I noticed now: there are typos here (missing uppercase F for flashArray and toekn).

**Comment by sdodsley** [comment]
@jsuchome @vuntz Final typos fixed. Ready for review again.
Thanks for agreeing to put this into Cloud 8. Our joint customers will be happy.

**Comment by sdodsley** [comment]
@vuntz @jsuchome Can someone give me a view of why the mkcloud check failed?

**Comment by jsuchome** [comment]
`neutron_lbaas.tests.tempest.v2.api.test_listeners_admin.ListenersTestJSON.test_create_listener_missing_tenant_id [3.676451s] ... FAILED`

That should not be related to your change; I'm gonna restart the build

**Comment by sdodsley** [comment]
@jsuchome Is this HA failure related to this PR?

**Pull request merged**